x/gov/abci.go

```diff
@@ -235,3 +267,42 @@ func safeExecuteHandler(ctx sdk.Context, msg sdk.Msg, handler baseapp.MsgService
 	res, err = handler(ctx, msg)
 	return
 }
+
+// failUnsupportedProposal fails a proposal that cannot be processed by gov
+func failUnsupportedProposal(
+	logger log.Logger,
+	ctx sdk.Context,
+	keeper *keeper.Keeper,
+	proposal v1.Proposal,
+	errMsg string,
+) error {
+	proposal.Status = v1.StatusFailed
+	proposal.FailedReason = fmt.Sprintf("proposal failed because it cannot be processed by gov: %s", errMsg)
+	proposal.Messages = nil // clear out the messages
+
+	if err := keeper.SetProposal(ctx, proposal); err != nil {
+		return err
+	}
+
+	if err := keeper.RefundAndDeleteDeposits(ctx, proposal.Id); err != nil {
+		return err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeInactiveProposal,
```

Comment:
should this event be same for proposals in `ActiveProposalsQueue`?

Reply:
Good one, updated 👍🏾